src/gui/MixerChannelView.cpp

```diff
@@ -0,0 +1,447 @@
+        const auto sendReceiveStateNone = !sendToThis && !receiveFromThis;
+
+        // Only one or none of them can be on
+        assert(sendToThis ^ receiveFromThis || sendReceiveStateNone);
+
+        m_sendArrow->setVisible(sendToThis);
+        m_receiveArrow->setVisible(receiveFromThis);
+
+        if (sendReceiveStateNone)
+        {
+            setSendReceiveState(SendReceiveState::None);
+        }
+        else
+        {
+            setSendReceiveState(sendToThis ? SendReceiveState::SendToThis : SendReceiveState::ReceiveFromThis);
+        }
+
+        QWidget::paintEvent(event);
+    }
+
+    void MixerChannelView::mousePressEvent(QMouseEvent*)
+    {
+        if (m_mixerView->currentMixerChannel() != this)
+        {
+            m_mixerView->setCurrentMixerChannel(this);
+        }
+    }
+
+    void MixerChannelView::mouseDoubleClickEvent(QMouseEvent*)
+    {
+        renameChannel();
+    }
+
+    bool MixerChannelView::eventFilter(QObject* dist, QEvent* event)
+    {
+        // If we are in a rename, capture the enter/return events and handle them
+        if (event->type() == QEvent::KeyPress)
+        {
+            auto keyEvent = static_cast<QKeyEvent*>(event);
+            if (keyEvent->key() == Qt::Key_Enter || keyEvent->key() == Qt::Key_Return)
+            {
+                if (m_inRename)
+                {
+                    renameFinished();
+                    event->accept(); // Stop the event from propagating
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    int MixerChannelView::channelIndex() const
+    {
+        return m_channelIndex;
+    }
+
+    void MixerChannelView::setChannelIndex(int index)
+    {
+        MixerChannel* mixerChannel = Engine::mixer()->mixerChannel(index);
+        m_fader->setModel(&mixerChannel->m_volumeModel);
+        m_muteButton->setModel(&mixerChannel->m_muteModel);
+        m_soloButton->setModel(&mixerChannel->m_soloModel);
+        m_effectRackView->setModel(&mixerChannel->m_fxChain);
+        m_channelIndex = index;
+    }
+
+    MixerChannelView::SendReceiveState MixerChannelView::sendReceiveState() const
+    {
+        return m_sendReceiveState;
+    }
+
+    void MixerChannelView::setSendReceiveState(const SendReceiveState& state)
+    {
+        m_sendReceiveState = state;
+        m_sendArrow->setVisible(state == SendReceiveState::SendToThis);
+        m_receiveArrow->setVisible(state == SendReceiveState::ReceiveFromThis);
+    }
+
+    QBrush MixerChannelView::backgroundActive() const
+    {
+        return m_backgroundActive;
+    }
+
+    void MixerChannelView::setBackgroundActive(const QBrush & c)
+    {
+        m_backgroundActive = c;
+    }
+
+    QColor MixerChannelView::strokeOuterActive() const
+    {
+        return m_strokeOuterActive;
+    }
+
+    void MixerChannelView::setStrokeOuterActive(const QColor & c)
```

Comment:
Extra space between type and `&`

Reply:
Fixed with commit 92878ce4030. Using reference/pointer at type like in other places in the code.